src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -1110,8 +1111,10 @@ private void InitializePackagesFolderInfo(string packagesFolderPath, bool exclud
                 var availablePackageDependencyInfoWithSourceSet = await ResolverGather.GatherAsync(gatherContext, token);
 
                 // emit gather dependency telemetry event and restart timer
-                TelemetryServiceUtility.EmitEventAndRestartTimer(telemetryHelper,
-                    string.Format(TelemetryConstants.GatherDependencyStepName, projectId));
+                TelemetryServiceUtility.StopTimer();
+                telemetryService?.EmitTelemetryEvent(
```

Comment:
Are we making sure that all these telemetry sub events will have the same operation id which is associated with the uber install or update operation?

Reply:
Yes, each telemetryService has unique operation id, each action will have their own service, so all sub events have the same operation id